fds-lambda-locations-ingestion/src/main/java/com/provectus/fds/ingestion/LocationsHandler.java

```diff
@@ -0,0 +1,203 @@
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.provectus.fds.models.events.Location;
+import com.provectus.fds.models.utils.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.net.URLDecoder;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+public class LocationsHandler implements RequestHandler<KinesisEvent, List<String>> {
+    private static final Logger logger = LoggerFactory.getLogger(LocationsHandler.class);
+
+    private static final String OK = "OK";
+    private static final String FAILED = "FAILED";
+
+    private static final String STREAM_NAME = "STREAM_NAME";
+    private static final String STREAM_NAME_DEFAULT = "locations";
+    private static final String AWS_REGION = "AWS_REGION";
+    private static final String AWS_REGION_DEFAULT = "us-west-2";
+    private static final String WRITE_BATCH_SIZE = "WRITE_BATCH_SIZE";
+    private static final String WRITE_BATCH_SIZE_DEFAULT = "100";
+
+    private static final String PREFIX = "locations/";
+    private static final String SEPARATOR = ",";
+    private static final int RETRY_COUNT = 3;
+
+    private final String streamName = System.getenv().getOrDefault(STREAM_NAME, STREAM_NAME_DEFAULT);
+    private final String region = System.getenv().getOrDefault(AWS_REGION, AWS_REGION_DEFAULT);
+    private final int batchSize = Integer.parseInt(System.getenv().getOrDefault(WRITE_BATCH_SIZE, WRITE_BATCH_SIZE_DEFAULT));
+
+    private final AmazonKinesis producer;
+    private ObjectMapper mapper;
+
+    public LocationsHandler() {
+        producer = getProducer();
+        mapper = new ObjectMapper();
+
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    }
+
+    @Override
+    public List<String> handleRequest(KinesisEvent event, Context context) {
+        List<String> results = new ArrayList<>();
+
+        for (KinesisEvent.KinesisEventRecord record : event.getRecords()) {
+            try {
+                S3Event s3Event = mapper.readerFor(S3Event.class)
+                        .readValue(record.getKinesis().getData().array());
+                results.add(handleRequest(s3Event));
+            } catch (IOException e) {
+                logger.error("Can't process S3 event", e);
+            }
+        }
+
+        return results;
+    }
+
+    private String handleRequest(S3Event s3Event) {
+        for (S3EventNotification.S3EventNotificationRecord record : s3Event.getRecords()) {
+            String bucket = record.getS3().getBucket().getName();
+            String key = record.getS3().getObject().getKey().replace('+', ' ');
```

Comment:
Is it URL decode? Is it enough? Maybe worth to consider some more powerful URL decoder?

Reply:
Removed it because it is not needed at all